numpy/core/einsumfunc.py

```diff
@@ -898,15 +900,19 @@ def einsum_path(*operands, optimize='greedy', einsum_call=False):
     naive_cost = _flop_count(indices, inner_product, len(input_list), dimension_dict)
 
     # Compute the path
-    if (path_type is False) or (len(input_list) in [1, 2]) or (indices == output_set):
+    if path is not None:
+        pass
+    elif (
+        (path_type is False)
+        or (len(input_list) in [1, 2])
+        or (indices == output_set)
+    ):
```

Comment:
Hmm, black strikes again :)  I don't like the placement of `:)`, but it is what it is. Maybe it will be improved some day.